File: java/src/main/java/com/factset/protobuf/stach/extensions/v2/ColumnOrganizedStachBuilder.java
package com.factset.protobuf.stach.extensions.v2;

import com.factset.protobuf.stach.extensions.ColumnStachExtensionBuilder;
import com.factset.protobuf.stach.extensions.StachExtensions;
import com.factset.protobuf.stach.v2.PackageProto;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.google.protobuf.InvalidProtocolBufferException;
import com.google.protobuf.util.JsonFormat;

public class ColumnOrganizedStachBuilder implements ColumnStachExtensionBuilder<PackageProto.Package> {

    private PackageProto.Package pkg;

    @Override
    public ColumnStachExtensionBuilder setPackage(PackageProto.Package pkg) {
        this.pkg = pkg;
        return this;
    }

    @Override
    public ColumnStachExtensionBuilder setPackage(Object pkgObject) throws JsonProcessingException, InvalidProtocolBufferException {
        ObjectMapper mapper = new ObjectMapper();
        String pkgString = mapper.writeValueAsString(pkgObject);
        return setPackage(pkgString);
    }

    @Override
    public ColumnStachExtensionBuilder setPackage(String pkgString) throws InvalidProtocolBufferException {

        PackageProto.Package.Builder builder = PackageProto.Package.newBuilder();
        JsonFormat.parser().ignoringUnknownFields().merge(pkgString, builder);
        this.pkg = builder.build();
        return this;
    }


    @Override
    public StachExtensions build() {
        return new ColumnOrganizedStachExtension(pkg);
    }
}

File: dotnet/StachExtensions/FactSet.Protobuf.Stach.Extensions/V2/ColumnOrganizedStachBuilder.cs
using Google.Protobuf;
using Newtonsoft.Json;

namespace FactSet.Protobuf.Stach.Extensions.V2
{
    public class ColumnOrganizedStachBuilder : IColumnOrganizedStachBuilder<Stach.V2.Package>
    {
        private Stach.V2.Package package;
        
        public IColumnOrganizedStachBuilder<Stach.V2.Package> SetPackage(Stach.V2.Package package)
        {
            this.package = package;
            return this;
        }

        public IColumnOrganizedStachBuilder<Stach.V2.Package> SetPackage(string package)
        {
            var jpSettings = JsonParser.Settings.Default;
            var jp = new JsonParser(jpSettings.WithIgnoreUnknownFields(true));
            this.package = jp.Parse<Stach.V2.Package>(package);
            return this;
        }

        public IColumnOrganizedStachBuilder<Stach.V2.Package> SetPackage(object package)
        {
            var pkgString = JsonConvert.SerializeObject(package);
            return SetPackage(pkgString);
        }

        public Stach.V2.Package GetPackage()
        {
            return package;
        }

        public IStachExtension Build()
        {
            return new ColumnOrganizedStachExtension(package);
        }
    }
}